core/scripts/chaincli/handler/handler.go

```diff
@@ -211,13 +220,54 @@ func (h *baseHandler) waitTx(ctx context.Context, tx *ethtypes.Transaction) erro
 	}
 
 	if receipt.Status == ethtypes.ReceiptStatusFailed {
+		// trying to extract failed reasons: https://ethereum.stackexchange.com/questions/48383/how-to-retrieve-revert-reason-for-past-transactions/80766#80766
+		reason, err := errorReason(ctx, h.client, h.fromAddr, tx, receipt.BlockNumber)
+		if err != nil {
+			log.Println("Failed to get error reason: ", err)
+		} else {
+			log.Println("Transaction failed. Reason: ", reason)
+		}
+
 		log.Println("Transaction failed: ", helpers.ExplorerLink(h.cfg.ChainID, tx.Hash()))
 		return errors.New("Transaction failed")
 	}
 
 	return nil
 }
 
+func errorReason(ctx context.Context, b ethereum.ContractCaller, from common.Address, tx *ethtypes.Transaction, blockNum *big.Int) (string, error) {
+	msg := ethereum.CallMsg{
+		From:     from,
+		To:       tx.To(),
+		Gas:      tx.Gas(),
+		GasPrice: tx.GasPrice(),
+		Value:    tx.Value(),
+		Data:     tx.Data(),
+	}
+	// execution reverted
+	res, err := b.CallContract(ctx, msg, blockNum)
+	if err != nil {
+		return "", fmt.Errorf("call contract error: %s", err.Error())
+	}
+	return unpackError(res)
+}
+
+var (
```

Comment:
Can we move this var block to the top of the file? I generally go for this order but I'm not super precious about it:

- imports 
- consts 
- vars 
- types
- constructors
- functions